lib/static/new-ui/features/suites/components/SuitesTreeView/utils.ts

```diff
@@ -238,12 +244,23 @@ export const sortTreeNodes = (entitiesContext: EntitiesContext, treeNodes: TreeN
                     return createWeight([treeNode.data.title.join(' '), testsCount, runsCount], {testsCount, runsCount});
                 } else if (currentSortExpression.type === SortType.ByFailedRuns) {
                     if (!childrenWeight) {
-                        notifyOfUnsuccessfulWeightComputation();
-                        return createWeight([0, 0, 0]);
+                        return createInvalidWeight(treeNode);
                     }
 
                     // For now, we assume there are no nested groups and suite/test weights are always 1 dimensional
                     return createWeight([childrenWeight.value[0], testsCount, runsCount], Object.assign({}, {testsCount, runsCount}, childrenWeight.metadata));
+                } else if (currentSortExpression.type === SortType.ByDuration) {
+                    if (!childrenWeight) {
+                        return createInvalidWeight(treeNode);
+                    }
+
+                    return childrenWeight;
+                } else if (currentSortExpression.type === SortType.ByStartTime) {
```

Comment:
I would either do one of these:
- pull `if (!childrenWeight)` check outside, between `if SortType.ByName` and `if SortType.ByFailedRuns` branches
- use `switch (currentSortExpression.type) case...`

We could also pull out functions `extractGroupWeight`, `extractSuiteWeight`, `extractBrowserWeight` in order to reduce nesting